docs: add control port section (#5164)
simplex-chat/simplex-chat: docs/SERVER.md

1. Generate two passwords for each user:

   ```sh
   tr -dc A-Za-z0-9 </dev/urandom | head -c 20; echo
   ```

2. Open the configuration file:

   ```sh
   vim /etc/opt/simplex/smp-server.ini
   ```

2. Configure the control port and replace the passwords:

   ```ini
   [AUTH]
   control_port_admin_password: <your_randomly_generated_admin_password>
   control_port_user_password: <your_randomly_generated_user_password>

   [TRANSPORT]
   control_port: 5224
   ```

3. Restart the server:

   ```sh
   systemctl restart smp-server
   ```

To access the control port, use:

```sh
nc 127.0.0.1 5224
```

or:

```sh
telnet 127.0.0.1 5224
```

Upon connecting, the control port should print:

```sh
SMP server control port
'help' for supported commands
```

To authenticate, type the following and hit enter. Change the `my_generated_password` with the `user` or `admin` password from the configuration:

```sh
auth my_generated_password
```

Here's the full list of commands, their descriptions and who can access them.

> | Command          | Description                                                                     | Can be accessed by         |

Reviewer: ```suggestion
| Command          | Description                                                                     | Requires `admin` role         |
```

| ---------------- | ------------------------------------------------------------------------------- | -------------------------- |
| `stats`          | Real-time statistics. Fields described in [Daily statistics](#daily-statistics) | `user`,`admin`             |
| `stats-rts`      | GHC/Haskell statistics. Can be enabled with `+RTS -T -RTS` option               | `user`,`admin`             |
| `clients`        | Clients information. Useful for debugging.                                      | `admin`                    |
| `sockets`        | General sockets information.                                                    | `user`,`admin`             |
| `socket-threads` | Thread infomation per socket. Useful for debugging.                             | `admin`                    |
| `threads`        | Threads information. Useful for debugging.                                      | `admin`                    |
| `server-info`    | Aggregated server infomation.                                                   | `user`,`admin`             |
| `delete`         | Delete known queue. Useful for content moderation.                              | `user`,`admin`             |

> | `save`           | Save queues/messages from memory.                                               | `user`,`admin`             |

Reviewer: ```suggestion
| `save`           | Save queues/messages from memory.                                               | `admin`             |
```

> | `stats`          | Real-time statistics. Fields described in [Daily statistics](#daily-statistics) | `user`,`admin`             |
> | `stats-rts`      | GHC/Haskell statistics. Can be enabled with `+RTS -T -RTS` option               | `user`,`admin`             |
> | `clients`        | Clients information. Useful for debugging.                                      | `admin`                    |
> | `sockets`        | General sockets information.                                                    | `user`,`admin`             |
> | `socket-threads` | Thread infomation per socket. Useful for debugging.                             | `admin`                    |
> | `threads`        | Threads information. Useful for debugging.                                      | `admin`                    |
> | `server-info`    | Aggregated server infomation.                                                   | `user`,`admin`             |
> | `delete`         | Delete known queue. Useful for content moderation.                              | `user`,`admin`             |
> | `save`           | Save queues/messages from memory.                                               | `user`,`admin`             |
> | `help`           | Help menu.                                                                      | `user`,`admin`             |
> | `quit`           | Exit the control port.                                                          | `user`,`admin`             |

Reviewer: ```suggestion
| `stats`          | Real-time statistics. Fields described in [Daily statistics](#daily-statistics) | -             |
| `stats-rts`      | GHC/Haskell statistics. Can be enabled with `+RTS -T -RTS` option               | -             |
| `clients`        | Clients information. Useful for debugging.                                      | yes                    |
| `sockets`        | General sockets information.                                                    | -             |
| `socket-threads` | Thread infomation per socket. Useful for debugging.                             | yes                    |
| `threads`        | Threads information. Useful for debugging.                                      | yes                    |
| `server-info`    | Aggregated server infomation.                                                   | -             |
| `delete`         | Delete known queue. Useful for content moderation.                              | -             |
| `save`           | Save queues/messages from memory.                                               | yes             |
| `help`           | Help menu.                                                                      | -             |
| `quit`           | Exit the control port.                                                          | -             |
```
